skxray/speckle_analysis.py

```diff
@@ -0,0 +1,311 @@
+        images sets
+        shapes is: (len(images_sets), )
+        one images_set is iterable of 2D arrays dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    mean_intensity_dict : dict
+        average intensity of each ROI as a dictionary
+        shape len(images_sets)
+        eg: 2 image sets,
+        {image set 1 : len(images in image set 1),
+        image set 2 : len(images in image set 2)}
+
+    index_list : list
+        labels list for each image set
+
+    """
+    mean_intensity_dict = {}
+    index_list = []
+    for n in range(len(images_set)):
+        mean_int, index = mean_intensity(images_set[n], labels)
+        mean_intensity_dict[n] = mean_int
+        index_list.append(index)
+
+    return mean_intensity_dict, index_list
+
+
+def mean_intensity(images, labels, index=None):
+    """
+    Mean intensities for ROIS' of the labeled array for set of images
+
+    Parameters
+    ----------
+    images : array
+        iterable of 2D arrays
+        dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    index : list
+        labels list
+        eg: 5 ROI's
+        index = [1, 2, 3, 4, 5]
+
+    Returns
+    -------
+    mean_intensity : array
+        mean intensity of each ROI for the set of images as an array
+        shape (len(images), number of labels)
+
+    """
+    if labels.shape != images[0].shape[0:]:
+        raise ValueError("Shape of the images should be equal to"
+                         " shape of the label array")
+    if index is None:
+        index = np.arange(1, np.max(labels) + 1)
+
+    mean_intensity = np.zeros((images.shape[0], index.shape[0]))
+    for n, img in enumerate(images):
+        mean_intensity[n] = mean(img, labels, index=index)
+
+    return mean_intensity, index
+
+
+def combine_mean_intensity(mean_int_dict, index_list):
+    """
+    Combine mean intensities of the images(all images sets) for each ROI
+    if the labels list of all the images are same
+
+    Parameters
+    ----------
+    mean_int_dict : dict
+        mean intensity of each ROI as a dictionary
+        eg: 2 image sets,
+        {image set 1 : (len(images in image set 1),
+        image set 2 : (len(images in image set 2)}
+
+    index_list : list
+        labels list for each image sets
+
+    Returns
+    -------
+    combine_mean_int : array
+        combine mean intensities of image sets for each ROI of labeled array
+        shape (len(images in all image sets), number of labels)
+
+    """
+    if np.all(map(lambda x: x == index_list[0], index_list)):
+        combine_mean_intensity = np.vstack(list(mean_int_dict.values()))
+    else:
+        raise ValueError("Labels list for the image sets are different")
+
+    return np.vstack(list(mean_int_dict.values()))
```

Comment:
and then do it again to return ?

The order of this will be arbitrary.  In python 2 it will be the same from run-to-run, but in python 3.4 this order will be random between python processes.